File: src/main/java/milk/entitymanager/entity/MagmaCube.php
package milk.entitymanager.entity;

import cn.nukkit.entity.Entity;
import cn.nukkit.event.entity.EntityDamageByEntityEvent;
import cn.nukkit.event.entity.EntityDamageEvent;

class MagmaCube extends Monster{
    const NETWORK_ID = 42;

    public width = 1.2;
    public height = 1.2;

    protected speed = 0.8;

    public function initEntity(){
        if(isset(this.namedtag.Health)){
            this.setHealth((int) this.namedtag["Health"]);
        }else{
            this.setHealth(16);
        }
        this.setMinDamage([0, 3, 4, 6]);
        this.setMaxDamage([0, 3, 4, 6]);
        super.initEntity();
        this.created = true;
    }

    public function getName(){
        return "MagmaCube";
    }

    public function attackEntity(Entity player){
        if(this.attackDelay > 10 && this.distanceSquared(player) < 1){
            this.attackDelay = 0;
            ev = new EntityDamageByEntityEvent(this, player, EntityDamageEvent.CAUSE_ENTITY_ATTACK, this.getDamage());
            player.attack(ev.getFinalDamage(), ev);
        }
    }

    public function getDrops(){
    	return [];
    }

}

File: src/main/java/milk/entitymanager/entity/MagmaCube.java
package milk.entitymanager.entity;

import cn.nukkit.entity.Entity;
import cn.nukkit.event.entity.EntityDamageByEntityEvent;
import cn.nukkit.event.entity.EntityDamageEvent;
import cn.nukkit.item.Item;
import cn.nukkit.level.format.FullChunk;
import cn.nukkit.nbt.tag.CompoundTag;

public class MagmaCube extends Monster{
    public static final int NETWORK_ID = 42;

    public MagmaCube(FullChunk chunk, CompoundTag nbt){
        super(chunk, nbt);
    }

    @Override
    public int getNetworkId(){
        return NETWORK_ID;
    }

    @Override
    public float getWidth(){
        return 1.2f;
    }

    @Override
    public float getHeight(){
        return 1.2f;
    }

    @Override
    public double getSpeed(){
        return 0.8;
    }

    public void initEntity(){
        super.initEntity();

        this.setDamage(new int[]{0, 3, 4, 6});
    }

    public String getName(){
        return "MagmaCube";
    }

    public void attackEntity(Entity player){
        if(this.attackDelay > 10 && this.distanceSquared(player) < 1){
            this.attackDelay = 0;

            EntityDamageEvent ev = new EntityDamageByEntityEvent(this, player, EntityDamageEvent.CAUSE_ENTITY_ATTACK, this.getDamage());
            player.attack(ev);
        }
    }

    public Item[] getDrops(){
    	return new Item[0];
    }

}
